src/puyapy/awst_build/pytypes.py

```diff
@@ -207,6 +207,35 @@ def wtype(self) -> typing.Never:
         raise CodeError(f"{self} is not usable as a value", self.source_location)
 
 
+def _flatten_nested_unions(types: Sequence[PyType]) -> Sequence[PyType]:
+    result = list[PyType]()
+    for t in types:
+        if isinstance(t, UnionType):
+            result.extend(t.types)
+        else:
+            result.append(t)
+    return unique(result)
+
+
+@typing.final
+@attrs.frozen
+class UnionType(PyType):
+    types: Sequence[PyType] = attrs.field(
+        converter=_flatten_nested_unions, validator=attrs.validators.min_len(1)
```

Comment:
When would a union with only a single type occur? Or should this be `min_len(2)`?

Reply:
It shouldn't no, and we probably do want a validation error rather than a code-error if it does, so yes, changed to 2.